samples/mpsl/timeslot/src/main.c

```diff
@@ -81,10 +100,8 @@ static mpsl_timeslot_signal_return_param_t *mpsl_timeslot_callback(
 				MPSL_TIMESLOT_SIGNAL_ACTION_REQUEST;
 		} else {
 			/* No return action, so timeslot will be ended */
-			signal_callback_return_param.params.request.p_next =
-				NULL;
 			signal_callback_return_param.callback_action =
-				MPSL_TIMESLOT_SIGNAL_ACTION_NONE;
+				MPSL_TIMESLOT_SIGNAL_ACTION_END;
 		}
 
```

Comment:
You should disable the TIMER0 interrupt, and clearing the CC event before ending the event

Reply:
updated